src/MCPServer/lib/utils.py

```diff
@@ -0,0 +1,18 @@
+
+import logging
+
+LOGGER = logging.getLogger('archivematica.mcp.server')
+
+def log_exceptions(fn):
+    """
+    Decorator to wrap a function in a try-catch that logs the exception.
+
+    Useful for catching exceptions in threads, which do not normally report back to the parent thread.
+    """
+    def wrapped(*args,  **kwargs):
```

Comment:
Good idea! functool.wraps doesn't seem to affect tracebacks though.  I can still add it but I don't know if it'll do anything useful.

The error looks like
```python
ERROR     2015-06-10 22:27:53  archivematica.mcp.server:utils:wrapped:18:  Uncaught exception
Traceback (most recent call last):
  File "~/archivematica/src/MCPServer/lib/utils.py", line 16, in wrapped
    return fn(*args, **kwargs)
  File "~/archivematica/src/MCPServer/lib/watchDirectory.py", line 74, in start
    raise Exception('uncaught')
Exception: uncaught
```